pkg/sif/descriptor_input.go

```diff
@@ -253,6 +253,23 @@ func OptSBOMMetadata(f SBOMFormat) DescriptorInputOpt {
 	}
 }
 
+// OptOCIBlobMetadata sets metadata for a OCIBlog data object.
+//
+// If this option is applied to a data object with an incompatible type, an error is returned.
+func OptOCIBlobMetadata(mediaType string) DescriptorInputOpt {
+	return func(t DataType, opts *descriptorOpts) error {
+		if got, want := t, DataOCIBlob; got != want {
+			return &unexpectedDataTypeError{got, []DataType{want}}
+		}
+
+		o := ociBlob{}
+		copy(o.MediaType[:], mediaType)
```

Comment:
It might be worth verifying that `mediaType <= len(o.MediaType)`, and returning an error in that case?